hail/src/main/scala/is/hail/expr/ir/TypeCheck.scala

```diff
@@ -251,6 +251,9 @@ object TypeCheck {
       case x@ArrayMap(a, name, body) =>
         assert(a.typ.isInstanceOf[TStreamable])
         assert(x.elementTyp == body.typ)
+      case x@ArrayZip(as, names, body, _) =>
+        assert(as.length == names.length)
+        assert(as.forall(_.typ.isInstanceOf[TStreamable]))
```

Comment:
Should this check `x.elementTyp == body.typ` like `ArrayMap` does? Or should `TypeCheck` not overlap with `InferType`?

Reply:
sure, good thing to check here I guess. It's a bit unclear what's the right code for typecheck vs infer.